### PR TITLE
Make copy link primary action and copy link with password secondary action

### LINKED ISSUE #11551: bring "Copy link and password" Split button
# Description

## User Stories

* > As a user who needs to work with a secure-by-default system I want a simple way to copy the link. 
## Acceptance Criteria
- provide split button "Copy link"
  - option: "Copy link and password"

## Definition of ready
- [ ] Everybody needs to understand the value written in the user story
- [ ] Acceptance criteria have to be defined
- [ ] All dependencies of the user story need to be identified
- [ ] Feature should be seen from an end user perspective
- [ ] Story has to be estimated
- [ ] Story points need to be less than 20

## Definition of done
- Functional requirements
  - [ ] Functionality described in the user story works
  - [ ] Acceptance criteria are fulfilled
- Quality
  - [ ] Code review happened
  - [ ] CI is green (that includes new and existing automated tests)
  - [ ] Critical code received unit tests by the developer
- Non-functional requirements
  - [ ] No sonar cloud issues

## Visual
screenshot_003123

### PR DESCRIPTION
**Pull request by AlexAndBear**
## Description
image

## Related Issue

- Fixes #11551 

## How Has This Been Tested?

- test environment:
- test case 1:
- test case 2:
- ...

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests
- [ ] Documentation
- [ ] Maintenance (e.g. dependency updates or tooling)

## Open tasks:

- [ ] ...

### CODE REVIEW

File: packages/web-pkg/src/components/CreateLinkModal.vue [hunk 1]
```diff
@@ -69,14 +69,47 @@
...
+    <div
+      class="link-modal-confirm-button-group oc-ml-s"
+      :class="{ 'oc-button-group': password.value }"
+    >
+      <oc-button
+        class="link-modal-confirm oc-modal-body-actions-confirm"
+        appearance="filled"
+        variation="primary"
+        :disabled="confirmButtonDisabled"
+        @click="onConfirm"
```
Comment (JammingBen) [inline packages/web-pkg/src/components/CreateLinkModal.vue:81]: Please keep using `$emit('confirm')` to make sure the confirm method of the modal wrapper gets called. It handles things like loading service and closing the modal afterwards. You can simply pass an argument for copying the password if needed.

File: packages/web-pkg/src/components/CreateLinkModal.vue [hunk 2]
```diff
@@ -268,12 +301,9 @@ export default defineComponent({
       }
 
       if (props.callbackFn) {
-        props.callbackFn({ result, password: password.value })
+        props.callbackFn({ result, password: password.value, options })
       }
     }
-
-    expose({ onConfirm })
```
Comment (JammingBen) [inline packages/web-pkg/src/components/CreateLinkModal.vue:275]: This needs to stay when working with `$emit` like suggested above.

File: packages/web-pkg/src/components/CreateLinkModal.vue (hunk 1 above)
Comment (AlexAndBear) [reply packages/web-pkg/src/components/CreateLinkModal.vue:81]: This ain't working for me :/ I literally cant pass the argument? Could you contribute ?

Comment (JammingBen) [reply packages/web-pkg/src/components/CreateLinkModal.vue:81]: Oh hm, let me try

Comment (JammingBen) [reply packages/web-pkg/src/components/CreateLinkModal.vue]: There was something missing in the modal wrapper component 😬 Now it should work!

Comment (AlexAndBear) [reply packages/web-pkg/src/components/CreateLinkModal.vue]: Ahhh thx big times =) !

**Review by JammingBen — APPROVED** [review]

**Pull request merged**